### DISCUSSION TITLE
Increase window padding when window count low

### DISCUSSION THREAD

**Discussion by yellingintothevoid** [asmvik/yabai#1419] [Ideas]
Since I'm using a 27" iMac, I don't need a single window to fill up the entire screen. To decrease window size, I've written a script that increases window padding when the number of windows is low. This is basically the same idea as in [this discussion](https://github.com/koekeishiya/yabai/discussions/804), but since it's more than a year old, I figured I'd open up a new one.
Using window padding to change window size introduces a noticeable delay when opening new windows though. Which is why I'd like to ask for help with my script or even a different method of achieving what I want. The delay is introduced by requiring four separate adjustments for padding (top, bottom, left, right). This means all windows get visibly shifted four times every time I open a new one. I'm very much a beginner to shell scripting and programming in general and I'm aware that my code is very likely to be pretty bad. I'm honestly quite surprised I've gotten this to work at all.

```
#!/bin/bash
currentsp=$(yabai -m query --spaces --space | jq '.index')
windowcount=$(yabai -m query --windows --space | jq -r '.[]."is-floating"' | grep false | wc -l)
excludes="Archive Utility\|Menuwhere\|Color Picker\|xQuartz\|VirtualBox WM"
countexcludes=$(yabai -m query --windows --space | jq -r '.[].app' | grep "${excludes}" | wc -l)
hiddenwindows=$(yabai -m query --windows --space | jq -r '.[]."is-visible"' | grep false | wc -l)
windowcount="$((windowcount - hiddenwindows - countexcludes))"

if [[ "$currentsp" == 5 ]]
	then
		yabai -m config \
			--space 5 top_padding    100 \
			--space 5 bottom_padding 100 \
			--space 5 left_padding   800 \
			--space 5 right_padding  800 
	else
		case $windowcount in
			[0-1])
				yabai -m config \
					--space mouse top_padding    100 \
					--space mouse bottom_padding 100 \
					--space mouse left_padding   950 \
					--space mouse right_padding  950 
				;;
			2)
				yabai -m config \
					--space mouse top_padding    150 \
					--space mouse bottom_padding 150 \
					--space mouse left_padding   500 \
					--space mouse right_padding  500 
				;;
			[3-4])
				yabai -m config \
					--space mouse top_padding    70 \
					--space mouse bottom_padding 70 \
					--space mouse left_padding   200 \
					--space mouse right_padding  200 
				;;
			5)
				yabai -m config \
					--space mouse top_padding    40 \
					--space mouse bottom_padding 40 \
					--space mouse left_padding   40 \
					--space mouse right_padding  40 
				;;
			*)
				yabai -m config \
					--space mouse top_padding    40 \
					--space mouse bottom_padding 40 \
					--space mouse left_padding   40 \
					--space mouse right_padding  40 
				;;
		esac
fi
```

Whether this can be improved or not, yabai is still an amazing window manager and I'm grateful for all the work and continual improvements that are being put into it!

**Comment by asmvik** [comment]
> The delay is introduced by requiring four separate adjustments for padding (top, bottom, left, right). This means all windows get visibly shifted four times every time I open a new one. 

I don't think you can script your way around that. Might have to either add some form of syntax to allow batching, or just introduce something like `all_padding` that sets the value for all directions.

**Reply by yellingintothevoid** [reply]
I appreciate that you'd be willing to implement something like that but i don't think it'd be necessary. The delay is noticeable but not really a hindrance or anything. I was hoping to maybe learn something new to give my script a little bit more polish, but I can totally live without it. Anyway, thank you for your answer! I'm excited about the new animations and am curious to see where all of this is going :)

**Reply by yellingintothevoid** [reply]
Hey, thanks for adding the option to set padding for all four directions simultaneously! I have updated my script and everything is so much faster now!

**Comment by Bellavene** [comment]
Thank you for that example!
